blueprints/to-locks.md

```diff
@@ -77,42 +77,43 @@ the CDN on behalf of user `A`.
 - GET+POST+DELETE
 
 Traffic Ops will need to add the logic to check for locks before snapping/ queueing a CDN. It'll also need to account for
-`shared` vs `exclusive` locks, and forbid a user from snapping/ queueing a CDN if another user possesses a lock on that CDN.
+`soft` vs `hard` locks, and forbid a user from snapping/ queueing a CDN if another user possesses a lock on that CDN.
+The following endpoints will need to handle the locks logic:
+- `/cachegroups/{{ID}}/queue_update`
+- `/cdns/{{ID}}/queue_update`
+- `/servers/{{hostname}}/queue_update`
+- `/snapshot`
+- `/topologies/{{name}}/queue_update`
```

Comment:
So, let me try to make sure I understand this one last time.

If I have a Soft lock on a CDN, anyone can still make changes, but these endpoints need to be aware that only a certain user is allowed to use them.

But if I have a Hard lock, these endpoints still need to be aware of that, but also nobody else is permitted to make any changes within the scope of the CDN, right? That means that very nearly every endpoint is affected; it might be a shorter list to itemize the endpoints that *won't* need to be changed as a result of this.

Basically, every endpoint that controls data scoped to a CDN (e.g. `/profiles`), every endpoint that controls data owned by objects scoped to a CDN (e.g. `/parameters`), and every endpoint that controls data that defines interactions between objects scoped to a CDN even if the data itself is not scoped to a CDN (e.g. `/cachegroups`) will need to be updated to be aware of the locking mechanism.

Reply:
You are correct. So basically all `PUT/POST/DELETE` endpoints that have any relation to a CDN should be locked down (to other users) if a user has a Hard lock on a particular CDN. I'll add this statement to the blueprint.